trunk/SUAVE/Analyses/Stability/AVL.py

```diff
@@ -0,0 +1,440 @@
+
+    """
+
+    def __defaults__(self):
+        self.tag        = 'avl'
+        self.keep_files = True
+
+        self.settings = Settings()
+
+        self.current_status = Data()
+        self.current_status.batch_index = 0
+        self.current_status.batch_file  = None
+        self.current_status.deck_file   = None
+        self.current_status.cases       = None
+        
+	self.settings.filenames.log_filename = sys.stdout
+        self.settings.filenames.err_filename = sys.stderr
+        
+        # Conditions table, used for surrogate model training
+        self.training = Data()        
+	
+	#HALE UAV Cruise
+	self.training.angle_of_attack  = np.array([0 , 2, 4, 6 ,8, 10]) * Units.deg
+	self.training.Mach             = np.array([0.075, 0.1, 0.125 ,0.15,0.175, 0.2])       
+    
+	# Standard subsonic/transolic aircarft
+	#self.training.angle_of_attack  = np.array([-2.,3.,8.])
+	#self.training.Mach             = np.array([0.3,0.7,0.85])  
+	
+        self.training.lift_coefficient = None
+        self.training.drag_coefficient = None
+        self.training_file             = None
+        
+        # Surrogate model
+        self.surrogates = Data()
+        self.surrogates.moment_coefficient = None
+	
+	
+	# Initialize quantities
+    
+	self.configuration = Data()    
+	self.geometry      = Data()
+    
+	self.stability_model                                = Data()
+	self.stability_model.short_period                   = Data()
+	self.stability_model.short_period.natural_frequency = 0.0
+	self.stability_model.short_period.damping_ratio     = 0.0
+	self.stability_model.phugoid                        = Data()
+	self.stability_model.phugoid.damping_ratio          = 0.0
+	self.stability_model.phugoid.natural_frequency      = 0.0
+	self.stability_model.roll_tau                       = 0.0
+	self.stability_model.spiral_tau                     = 0.0 
+	self.stability_model.dutch_roll                     = Data()
+	self.stability_model.dutch_roll.damping_ratio       = 0.0
+	self.stability_model.dutch_roll.natural_frequency   = 0.0
+	
+
+    def finalize(self):
+
+        geometry           = self.geometry
+        self.tag           = 'avl_analysis_of_{}'.format(geometry.tag)
+	configuration      = self.configuration
+	stability_model    = self.stability_model
+    
+	configuration.mass_properties = geometry.mass_properties
+    
+	if geometry.has_key('fuel'): #fuel has been assigned(from weight statements)
+	    configuration.fuel = geometry.fuel
+	else: #assign as zero to allow things to run
+	    fuel                       = SUAVE.Components.Physical_Component()
+	    fuel.mass_properties.mass  = 0.
+	    configuration.fuel         = fuel	
+
+
+        run_folder = self.settings.filenames.run_folder 
+        
+        # Sample training data
+        self.sample_training()
+    
+        # Build surrogate
+        self.build_surrogate()
+
+        return
+
+    def __call__(self,conditions):
+
+        # Unpack
```

Comment:
I think working months ago in Spyder changed the indentation. I think it has been rectified